src/index.test.ts

```diff
@@ -1,5 +1,177 @@
+import { mocked } from 'ts-jest/utils';
+
+import { readableFromArray } from './test-helpers/stream-utils';
+import * as express from 'express';
+import { Application } from 'express-serve-static-core';
+import * as request from 'supertest';
+import {
+  lookupDomain,
+  getSiteById,
+  IHubRequestOptions,
+} from '@esri/hub-common';
+import * as config from 'config';
+jest.mock('config');
+
+// this fancy code is just to _only_ mock some fns
+// and leave the rest alone
+jest.mock('@esri/hub-common', () => ({
+  ...(jest.requireActual('@esri/hub-common') as object),
+  getSiteById: jest.fn(),
+  lookupDomain: jest.fn()
+}));
+
+import * as mockDomainRecord from './test-helpers/mock-domain-record.json';
+import * as mockSiteModel from './test-helpers/mock-site-model.json';
+import * as mockDataset from './test-helpers/mock-dataset.json';
+import { IItem } from '@esri/arcgis-rest-types';
+
+const Output = require('./index');
+
 describe('Output Plugin', () => {
-  it('has correct type', () => {
-    expect(true).toBeTruthy();
+  const mockConfigModule = mocked(config, true);
+  const mockLookupDomain = mocked(lookupDomain);
+  const mockGetSite = mocked(getSiteById);
+  let plugin;
+  let app: Application;
+
+  const siteHostName = 'download-test-qa-pre-a-hub.hubqa.arcgis.com';
+
+  beforeEach(() => {
+    mockLookupDomain.mockResolvedValue(mockDomainRecord);
+    mockGetSite.mockResolvedValue(mockSiteModel);
+
+    plugin = new Output();
+    plugin.model = {
+      pullStream: jest.fn().mockResolvedValue(readableFromArray([mockDataset])),
+    };
+
+    app = express();
+    app.get('/dcat', plugin.serve.bind(plugin));
+
+    mockConfigModule.get.mockReturnValue('foobar');
+  });
+
+  it('is configured correctly', () => {
+    expect(Output.type).toBe('output');
+    expect(Output.version).toBeDefined();
+    expect(Output.routes).toEqual([
+      {
+        path: '/dcat-ap/2.0.1',
+        methods: ['get'],
+        handler: 'serve',
+      },
+    ]);
+  });
+
+  it('handles a DCAT request', async () => {
+    await request(app)
```

Comment:
Opted for the `supertest` route to flex the handler as realistically as possible.